pkg/systemd/quadlet/quadlet.go

```diff
@@ -268,6 +269,18 @@ func usernsOpts(kind string, opts []string) string {
 	return res.String()
 }
 
+func parseUserGroup(in string) string {
+	if strings.HasPrefix(in, "%") {
+		return in
+	}
+
+	if _, err := strconv.ParseUint(in, 10, 32); err == nil {
+		return in
+	}
+
+	return "0"
```

Comment:
Shouldn't we return a proper error, silently defaulting to 0 seems like an odd choice.

Reply:
Also why does quadlet not accept names, `--user` accepts names assuming they are present in `/etc/passwd` and `/etc/group`
```
podman run --user bin:nobody alpine id
uid=1(bin) gid=65534(nobody) groups=65534(nobody)
```

Reply:
> Shouldn't we return a proper error, silently defaulting to 0 seems like an odd choice.

That's where it gets kinda muddy, do you want to double validate every value in `quadlet` or just pass it directly to `podman`? 
- To me, the less `quadlet` gets in the way, the better the user experience will be because ultimately, you'll end up using `journalctl --user -fexu container.service` when things break. 
- And currently for `quadlet` errors, you have to use the `-dryrun` flag and pipe stdout to avoid getting hit with a wall of text. I got another PR lined up which adds a `-target` flag so you can specify a single file to test / check: https://github.com/devnode/podman/commit/bd03a41ce3ea36e3052d5c78869efebc1b645121

See also:

[LookupUID](https://github.com/containers/podman/blob/911be1cbcb0ad654a8e10666262bdcee50ee54e0/pkg/systemd/parser/unitfile.go#L692)
[LookupGID](https://github.com/containers/podman/blob/911be1cbcb0ad654a8e10666262bdcee50ee54e0/pkg/systemd/parser/unitfile.go#L725)

Reply:
I agree that Quadlet should not validate what podman validates unless the validation is trivial. Now, adding @Luap99's comment about strings, it seems that all you need to do is to remove the conversion to integer and pass the values as they are

Reply:
> pass the values as they are

That's likely the safest way forward :+1:

Reply:
> it seems that all you need to do is to remove the conversion to integer and pass the values as they are

Should we throw an error (or maybe a deprecation notice, lol) if the `Group` key is provided without a `User` or just silently ignore it? 

Currently, it'll return `--user 0:1000`